src/lib/y2storage/device.rb

```diff
@@ -46,5 +51,49 @@ class Device
     storage_forward :remove_descendants
     storage_forward :userdata
     storage_forward :userdata=
+
+    # Ancestors in the devicegraph, not including the device itself
+    #
+    # @note: this is slightly different from Storage::Device#ancestors, which
+    # requires an argument to decide if the device itself should be included in
+    # the result.
+    #
+    # @return [Array<Device>]
+    def ancestors
+      itself = false
+      storage_ancestors(itself)
+    end
+
+    # Checks whether the device is a concrete kind(s) of device.
+    #
+    # Always false for this base class, which represents an abstract device.
+    # To be redefined by classes representing more concrete devices.
+    #
+    # The goal of this method is to provide a more convenient alternative to the
+    # usage of Object#is_a? that doesn't rely on fully qualified class names and
+    # that can be extended or customized by each device subclass. See examples.
+    #
+    # @example Checking if a device is a disk
+    #
+    #   encryption.blk_device.is?(:disk)
+    #
+    # @example Filtering disks and partitions
+    #
+    #   something.ancestors.select { |dev| dev.is?([:disk, "partition"]) }
+    #
+    # @param types [#to_sym, Array<#to_sym>] name (or names) of the device
+    #   type, as defined in each subclass
+    # @return [Boolean]
+    def is?(_types)
```

Comment:
thinking about it, I do not like much this sym or array and prefer `*types` so usage will be 
```
encryption.blk_device.is?(:disk)

something.ancestors.select { |dev| dev.is?(:disk, "partition") }
```